File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        int index = 0;
        while(index < s.length()){
            String word = s.substring(index,index+1);
            int wcount =1;
            int dwcount =0;
            index++;
            while(wcount !=dwcount){
                if(index>=s.length()){
                    break;
                }
                if(s.substring(index, index+1).equals(word)){
                    wcount++;
                }else{
                    dwcount++;
                }
                index++;
            }
            answer++;
            
        }
        return answer;
    }
}

File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    s=list(s)
    wcount=1
    dwcount=0
    index=1
    while len(s)>=1: # sss aaaaaabcdf
        if wcount+dwcount >=len(s):
            answer+=1
            break            
        if wcount==dwcount:
            answer+=1
            del s[0:wcount + dwcount]
            index=0
            wcount=1
            dwcount=0
        else:
            if s[0]==s[index]:
                wcount+=1
            else:
                dwcount+=1
                
        if len(s)>1:
            index+=1
        else:
            answer+=1
            break
    
    return answer